packages/blade/src/components/BottomSheet/_decisions/decisions.md

```diff
@@ -0,0 +1,245 @@
+
+
+### Composition Example
+
+```jsx
+import { 
+  useTheme,
+  useBreakpoint,
+  BottomSheet,
+  Dropdown
+} from "@razorpay/blade";
+
+const App = () => {
+  const { theme } = useTheme();
+  const { matchedDeviceType } = useBreakpoint({ breakpoints: theme.breakpoints });
+  const isMobile = matchedDeviceType === 'mobile';
+
+  return (
+    <Dropdown selectionType={selectionType}>
+      <SelectInput label="Select Action" />
+      {isMobile ? (
+        <BottomSheet>
+          <BottomSheetHeader />
+          <BottomSheetBody>
+            <SelectContent />
+          </BottomSheetBody>
+          <BottomSheetFooter />
+        </BottomSheet>
+      ) : (
+        <DropdownOverlay>
+          <SelectContent />
+        </DropdownOverlay>
+      )}
+    </Dropdown>
+  );
+};
+
+const SelectContent = () => {
+  return (
+    <ActionList>
+      <ActionListItem
+        leading={<ActionListItemIcon icon={SettingsIcon} />}
+        title="Settings"
+        value="settings"
+      />
+      <ActionListItem leading={<ActionListItemIcon icon={InfoIcon} />} title="Info" value="info" />
+    </ActionList>
+  );
+};
+
+```
+
+**Q.** Why can't we lazy load from blade side? 
+
+There are two major reasons: 
+
+
+1. It's generally not feasible to lazy load components from library side, We can't possibly know where & how a blade component can be used or imported in the user's codebase. Even if we know that it will also require changes in how we bundle blade, since right now blade distributes the bundle in single index.js chunks, introducing lazy loading will create more fragmented chunks. 
+
+2. There are multiple ways to lazy load react components and it depends on the app's architecture. For example, React.lazy only does client side lazy loading while loadable support SSR too, thus the descision needs to be handed over to the consumers to import & integrate the components as they fit on their stack.
+
+
+## Accessibility
+
+The bottom sheet will follow the accessibility characteristics of a Modal.
+
+[APG Guidelines for Modal](https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/)
+
+Behaviours:
+
+- Tab, Shift+Tab should cycle the focused elements
+- The focus should be trapped inside the bottom sheet
+- Escape closes the bottom sheet, after closing the focus will go to the triggered element
+- When opening the bottom sheet it should either focus on the first focusable element on the sheet or element provided by the user via `initialFocusRef` prop
+
+## Open questions
+
+1. What is `action` in the header trailing visual?
+2. In BottomSheetHeaderLeaing & BottomSheetFooterLeading will the `prefix` only support Icon component?
+3. In BottomSheetBody will we only have ActionList or users can add any of their own components too? 
```

Comment:
this we can't restrict for cases like where people expect input elements for OTP?

Reply:
This was old open question on the first draft, removed. we can have anything.